server/src/main/java/org/opensearch/action/search/SearchRequestSlowLog.java

```diff
@@ -0,0 +1,274 @@
+        clusterService.getClusterSettings()
+            .addSettingsUpdateConsumer(CLUSTER_SEARCH_REQUEST_SLOWLOG_THRESHOLD_TRACE_SETTING, this::setTraceThreshold);
+        clusterService.getClusterSettings().addSettingsUpdateConsumer(CLUSTER_SEARCH_REQUEST_SLOWLOG_LEVEL, this::setLevel);
+    }
+
+    @Override
+    public void onPhaseStart(SearchPhaseContext context) {}
+
+    @Override
+    public void onPhaseEnd(SearchPhaseContext context, SearchRequestContext searchRequestContext) {
+        long tookInNanos = System.nanoTime() - context.getCurrentPhase().getStartTimeInNanos();
+        searchRequestContext.updatePhaseTookMap(context.getCurrentPhase().getName(), TimeUnit.NANOSECONDS.toMillis(tookInNanos));
+    }
+
+    @Override
+    public void onPhaseFailure(SearchPhaseContext context) {}
+
+    @Override
+    public void onRequestStart(SearchRequestContext searchRequestContext) {
+        searchRequestContext.setAbsoluteStartNanos(System.nanoTime());
+    }
+
+    @Override
+    public void onRequestEnd(SearchPhaseContext context, SearchRequestContext searchRequestContext) {
+        long tookInNanos = System.nanoTime() - searchRequestContext.absoluteStartNanos();
+
+        if (warnThreshold >= 0 && tookInNanos > warnThreshold && level.isLevelEnabledFor(SlowLogLevel.WARN)) {
+            logger.warn(new SearchRequestSlowLogMessage(context, tookInNanos, searchRequestContext));
+        } else if (infoThreshold >= 0 && tookInNanos > infoThreshold && level.isLevelEnabledFor(SlowLogLevel.INFO)) {
+            logger.info(new SearchRequestSlowLogMessage(context, tookInNanos, searchRequestContext));
+        } else if (debugThreshold >= 0 && tookInNanos > debugThreshold && level.isLevelEnabledFor(SlowLogLevel.DEBUG)) {
+            logger.debug(new SearchRequestSlowLogMessage(context, tookInNanos, searchRequestContext));
+        } else if (traceThreshold >= 0 && tookInNanos > traceThreshold && level.isLevelEnabledFor(SlowLogLevel.TRACE)) {
+            logger.trace(new SearchRequestSlowLogMessage(context, tookInNanos, searchRequestContext));
+        }
+    }
+
+    /**
+     * Search request slow log message
+     *
+     * @opensearch.internal
+     */
+    static final class SearchRequestSlowLogMessage extends OpenSearchLogMessage {
+
+        SearchRequestSlowLogMessage(SearchPhaseContext context, long tookInNanos, SearchRequestContext searchRequestContext) {
+            super(prepareMap(context, tookInNanos, searchRequestContext), message(context, tookInNanos, searchRequestContext));
+        }
+
+        private static Map<String, Object> prepareMap(
+            SearchPhaseContext context,
+            long tookInNanos,
+            SearchRequestContext searchRequestContext
+        ) {
+            final Map<String, Object> messageFields = new HashMap<>();
+            messageFields.put("took", TimeValue.timeValueNanos(tookInNanos));
+            messageFields.put("took_millis", TimeUnit.NANOSECONDS.toMillis(tookInNanos));
```

Comment:
I get that this was copied from `SearchSlowLog`, but I don't think we need to continue to emit both nanos and millis. Can we just emit nanos?

Reply:
They are not just milli and nano. `took_millis` is the bare value in millis (ex "10000") but `took` is human readable with unit (ex "10s", "100micros", "15nanos").

Reply:
I am favor of keep this consistent (keep or remove) at both the places. Anyway from David's point above, it does make sense as is